### PR TITLE
DPTP-2920: Granting ClusterEditor role to specific users in OSD clusters

### PR DESCRIPTION
**Pull request by apodhrad**
Signed-off-by: Andrej Podhradsky <apodhrad@redhat.com>

### CODE REVIEW

**Comment by apodhrad** [comment]
Hi @bradmwilliams @jupierce could you have a look at this PR, please?

File: ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    fi
+    # Grant ClusterEditor role on OSD via ocm
+    ocm_grant_cluster_editor_role "${cluster_id}" "${cluster_editor}"
+    # Share the appropriate secret on openshift via oc
+    oc_share_cluster_editor_token "${kubeconfig}" "${cluster_editor}"
+  fi
+}
+
+# Sourceable from *test.sh files
+if [[ ${0} != *"test.sh" ]]; then
```
Comment (samvarankashyap) [inline ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh:178]: @apodhrad What is the functionality of test.sh ? 
Where is it inside the PR?

Comment (samvarankashyap) [reply ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh:178]: Further, should we be providing an option to customers to skip this test all together ?

**Comment by bradmwilliams** [comment]
Hey @apodhrad,
Since this PR's has been `/approved`, anyone in the Openshift organization should be able to `/lgtm` it.  I would recommend having someone on your team take a look and if everything is in order, then they can tag it for you.

File: ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh (hunk 1 above)
Comment (apodhrad) [reply ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh:178]: Hi @samvarankashyap , I have an external test script (with suffix `*test.sh`) which test particular functions (unit tests). Unfortunately, PR checks failed when I added the test script there :(

I planned to investigate the issue but until that I wanted someone to review this PR.

Comment (apodhrad) [reply ci-operator/step-registry/osd/grant/cluster-editor/osd-grant-cluster-editor-commands.sh:178]: IMHO, the part
```
if [[ ${0} != *"test.sh" ]];
```
can be kept as it skips the auto execution of the main functionality if and only if `osd-grant-cluster-editor-commands.sh` is sources from a file with suffix `*test.sh`.

**Comment by apodhrad** [comment]
Hi @samvarankashyap I have added the test script and it seems there are no issues with it

**Comment by apodhrad** [comment]
The test can be found at
https://gitlab.cee.redhat.com/apodhrad/release-tests

**Pull request merged**